Firestore/Swift/Source/Codable/CodablePassThroughTypes.swift

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import FirebaseFirestore
+
+func isCodablePassThroughType<T: Any>(_ value: T) -> Bool {
```

Comment:
If it's `Codable`, then both the Encodable and Decodable calls will fail to match because of not conforming to the other. Making changes 1 and 3.

```
/Users/paulbeusterien/gh/firebase-ios-sdk/Firestore/Swift/Source/Codable/third_party/FirestoreEncoder.swift:345:42: error: argument type 'T' does not conform to expected type 'Decodable'

/Users/paulbeusterien/gh/firebase-ios-sdk/Firestore/Swift/Source/Codable/third_party/FirestoreDecoder.swift:1008:43: error: argument type 'T' does not conform to expected type 'Encodable'
```